plone/formwidget/namedfile/utils.py

```diff
@@ -39,15 +39,18 @@ def upload_map(self):
         upload_map = annotations.setdefault(FILE_UPLOAD_MAP_KEY, OOBTree())
         return upload_map
 
-    def cleanup(self):
+    def cleanup(self, force=False):
         """Remove obsolete temporary uploads.
+
+        To avoid conflict errors, files are deleted on every ~5th method call.
+        Use force to make sure all expired files are deleted
         """
         upload_map = self.upload_map
-        for key, val in upload_map.items():
-            if val.get('dt', FALLBACK_DATE) < (
-                datetime.now() - timedelta(
-                    seconds=FILE_UPLOAD_EXPIRATION_TIME
-                )
-            ) and randint(0, 5) == 0:  # Avoid conflict errors by deleting only every fifth time  # noqa
+        expiration_limit = datetime.now() - timedelta(seconds=FILE_UPLOAD_EXPIRATION_TIME)
+        # Avoid conflict errors by deleting only every fifth time
+        delete = force or randint(0, 5) == 0
```

Comment:
This is every sixth time: "random integer N such that a <= N <= b"
https://docs.python.org/3/library/random.html#random.randint
I would declare a global variable at the top of the file and use it, like:
```suggestion
        delete = force or randint(0, CLEANUP_INTERVAL-1) == 0
```

Reply:
good point. implemented the change